ledger/src/blockstore_processor.rs

```diff
@@ -154,6 +154,20 @@ fn aggregate_total_execution_units(execute_timings: &ExecuteTimings) -> u64 {
     execute_cost_units
 }
 
+fn sum_additional_costs(batch: &TransactionBatch) -> u64 {
```

Comment:
probably want to benchmark the performance impact of adding this function during replay.

Reply:
> Is the motivation for this change elaborated in an issue somewhere?

As i understand it, the main concern is that a leader could choose to put more transactions into a block than it should, which would increase its profits (as long as they didn't overdo it). Essentially overclocking the system, which would work so long as enough machines could still finish in time and reach consensus. This validator-side check would prevent that hack from being profitable.